src/index.js

```diff
@@ -1,52 +1,125 @@
+      const files = (opts.userData || {}).files;
+
+      // TODO: This is a hack for simulator. Figure out how to load plugin state
+      if (!features && opts.userData.__FEATURES__) {
+        features = opts.userData.__FEATURES__;
+      }
+
+      if (!Array.isArray(features) || !features.length) {
+        return Promise.reject(new Error('Expected RoI features inputs'));
+      } else if (features.some(f => FreeSurfer.validFields.indexOf(f) < 0)) {
+        return Promise.reject(new Error(
+          `Invalid FreeSurfer feature in ${features.toString()}`
+        ));
+      } else if (!Array.isArray(files) || !files.length) {
+        return Promise.reject(new Error('Expected user data to contain files'));
+      }
+
+      const pickFeature = getFreeSurferDataPicker(features[0]);
+
+      return Promise.all(files.map(f => readFileAsync(f.filename)))
+        .then(freeSurferDatas => ({
+          X: files.map(getNormalizedTags), // covariates
+          y: freeSurferDatas.map(pickFeature), // RoIs
+        }));
+    },
+    inputs: [{
+      help: 'Select Freesurfer region(s) of interest',
+      label: 'Freesurfer ROI',
+      type: 'select',
+      values: FreeSurfer.validFields,
+    }, {
+      label: 'Covariates',
+      type: 'covariates',
+    }],
+  }, {
     type: 'cmd',
     cmd: 'python',
     args: ['./ridge_regress.py'],
     verbose: true,
-  },
+  }],
   remote: {
     type: 'function',
     fn(opts) {
-      const data = opts.previousData || {};
-      opts.userResults.forEach(rslt => (data[rslt.username] = rslt.data));
-      if (data && Object.keys(data).length === opts.usernames.length) {
-        data.complete = true;
-      }
-      console.error(data); // eslint-disable-line
-      let beta=opts.userResults.map(r=>r.data);
-//      let intercept=beta.map(r=>r.intercept);
-      let beta_vector=beta.map(r=>r.beta_vector);
-      console.log(beta_vector)
-      
-//      var sum_intercept = 0;
-      var sum_beta_vector = new Array(beta_vector[0].length).fill(0);
-
-      for( var i = 0; i < beta.length; i++ ){
-//         sum_intercept += intercept[i];
-         for( var j=0; j < beta_vector[0].length; j++){
-       
-             sum_beta_vector[j] += beta_vector[i][j]; //don't forget to add the base
-         }
-      }
-      console.log(sum_beta_vector)  
-//      var avg_intercetp = sum_beta0/beta.length;
+      const userResults = opts.userResults;
 
-     var avg_beta_vector = new Array(beta_vector[0].length).fill(0);
-     for( var i=0; i<beta_vector[0].length; i++){
-        
-         avg_beta_vector[i] = sum_beta_vector[i]/beta.length;
+      // Not all user results contain betas. Return early.
+      if (userResults.some(userResult => !((userResult || {}).data || {}).beta_vector)) {
+        return {};
       }
 
-      console.log('done! the average beta_vector is',avg_beta_vector);
-      return data;
+      return {
+        averageBetaVector: userResults[0].data.beta_vector.reduce(
```

Comment:
It’s important to return the average beta vector. Otherwise, COINSTAC can’t save it!